cli/ts/commands/joinPoll.ts

```diff
@@ -0,0 +1,461 @@
+import { type ContractTransactionReceipt } from "ethers";
+import { extractVk, genProof, verifyProof } from "maci-circuits";
+import { formatProofForVerifierContract, genSignUpTree, IGenSignUpTree } from "maci-contracts";
+import { MACI__factory as MACIFactory, Poll__factory as PollFactory } from "maci-contracts/typechain-types";
+import { CircuitInputs, IJsonMaciState, MaciState, IPollJoiningCircuitInputs } from "maci-core";
+import { poseidon, sha256Hash, stringifyBigInts } from "maci-crypto";
+import { IVkObjectParams, Keypair, PrivKey, PubKey, StateLeaf } from "maci-domainobjs";
+
+import assert from "assert";
+import fs from "fs";
+
+import type { IJoinPollArgs, IJoinedUserArgs, IParsePollJoinEventsArgs, IJoinPollData } from "../utils";
+
+import { contractExists, logError, logYellow, info, logGreen, success } from "../utils";
+import { banner } from "../utils/banner";
+
+/**
+ * Get state index and credit balance
+ * either from command line or
+ * from maci state leaves or from sign up leaves
+ * @param stateIndex State index from the command
+ * @param newVoiceCreditBalance Credit balance from the command
+ * @param stateLeaves State leaves from maci state or sign up tree
+ * @param userMaciPubKey Public key of the maci user
+ * @returns State index and credit balance
+ */
+const getStateIndexAndCreditBalance = (
+  stateIndex: bigint | null,
+  newVoiceCreditBalance: bigint | null,
+  stateLeaves: StateLeaf[],
+  userMaciPubKey: PubKey,
+) => {
+  let loadedStateIndex = stateIndex;
+  let loadedCreditBalance = newVoiceCreditBalance;
+
+  if (!stateIndex) {
+    const index = stateLeaves.findIndex((leaf) => leaf.pubKey.equals(userMaciPubKey));
+    if (index) {
+      loadedStateIndex = BigInt(index);
+    } else {
+      logError("State leaf not found");
+      process.exit();
+    }
```

Comment:
It won't work if there is no such element in stateLeaves. `findIndex` returns `-1` in this case and error will be thrown for `0` index.

Reply:
https://github.com/0x3327/maci/commit/8134d57c11a820f4001e180bfcb29b5e1ec20012